homeassistant/components/landisgyr_heat_meter/const.py

```diff
@@ -0,0 +1,192 @@
+        key="volume_usage_m3",
+        icon="mdi:fire",
+        name="Volume usage",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    # Diagnostic entity for debugging, this will match the value in GJ indicated on the meter's display
+    SensorEntityDescription(
+        key="heat_usage_gj",
+        icon="mdi:fire",
+        name="Heat usage GJ",
+        native_unit_of_measurement="GJ",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="heat_previous_year",
+        icon="mdi:fire",
+        name="Heat usage previous year",
+        native_unit_of_measurement=ENERGY_MEGA_WATT_HOUR,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="volume_previous_year_m3",
+        icon="mdi:fire",
+        name="Volume usage previous year",
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="ownership_number",
+        name="Ownership number",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="error_number",
+        name="Error number",
+        icon="mdi:home-alert",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="device_number",
+        name="Device number",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="measurement_period_minutes",
+        name="Measurement period minutes",
+        icon="mdi:clock-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="power_max_kw",
+        name="Power max",
+        native_unit_of_measurement="kW",
+        icon="mdi:power-plug-outline",
+        device_class=SensorDeviceClass.POWER,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="power_max_previous_year_kw",
+        name="Power max previous year",
+        native_unit_of_measurement="kW",
+        icon="mdi:power-plug-outline",
+        device_class=SensorDeviceClass.POWER,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="flowrate_max_m3ph",
+        name="Flowrate max",
+        native_unit_of_measurement="m3ph",
+        icon="mdi:water-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="flowrate_max_previous_year_m3ph",
+        name="Flowrate max previous year",
+        native_unit_of_measurement="m3ph",
+        icon="mdi:water-outline",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="return_temperature_max_c",
+        name="Return temperature max",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
```

Comment:
We don't need a custom icon when setting a device class normally. The device class will show with a specific icon in the frontend already.